content/en/blog/_posts/2024-08-15-moving-cgroup-v1-support-maintenance-mode-kubernetes-1-31.md

```diff
@@ -0,0 +1,100 @@
+## Cgroups in Kubernetes
+
+Kubernetes relies heavily on cgroups to manage and isolate the resources consumed
+by containers running in pods. Each container in Kubernetes is placed in its own
+cgroup, which allows Kubernetes to enforce resource limits, monitor usage, and
+ensure fair resource distribution among all containers.
+
+### How Kubernetes Uses cgroups:
+
+* **Resource Allocation**
+: Ensures that containers do not exceed their allocated CPU and memory limits.
+
+* **Isolation**
+: Isolates containers from each other to prevent resource contention.
+
+* **Monitoring**
+: Tracks resource usage for each container to provide insights and metrics.
+
+## Transitioning to Cgroup v2
+
+The Linux community has been focusing on cgroup v2 for new features and
+improvements. Major Linux distributions and projects like
+[systemd](https://github.com/systemd/systemd/issues/30852) are
+transitioning towards cgroup v2, which provides several benefits over cgroup
+v1, such as Unified Hierarchy, Improved Interface, Better Resource Control,
+[cgroup aware OOM killer](https://github.com/kubernetes/kubernetes/pull/117793),
+[rootless support](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2033-kubelet-in-userns-aka-rootless/README.md#cgroup) etc.
+
+Given these advantages, Kubernetes is also making the move to embrace cgroup
+v2 more fully. However, this transition needs to be handled carefully to avoid
+disrupting existing workloads and to provide a smooth migration path for users.
+
+## Moving cgroup v1 Support into Maintenance Mode
+
+### What Does Maintenance Mode Mean?
+
+When cgroup v1 is placed into maintenance mode in Kubernetes, it means that:
+
+1. **Feature Freeze**: No new features will be added to cgroup v1 support.
+2. **Security Fixes**: Critical security fixes will still be provided.
+3. **Best-Effort Bug Fixes**: Major bugs may be fixed if feasible, but some
+issues might remain unresolved.
+
+### Why Move to Maintenance Mode?
+
+The move to maintenance mode is driven by the need to stay in line with the
+broader ecosystem and to encourage the adoption of cgroup v2, which offers
+better performance, security, and usability. By transitioning cgroup v1 to
+maintenance mode, Kubernetes can focus on enhancing support for cgroup v2
+and ensure it meets the needs of modern workloads. It's important to note
+that maintenance mode does not mean deprecation; cgroup v1 will continue to
+receive critical security fixes and major bug fixes as needed.
+
+## Impact on Users
+
+Users currently relying on cgroup v1 are highly encouraged to plan for the
+transition to cgroup v2. This transition involves:
+
+1. **Upgrading Systems**: Ensuring that the underlying operating systems and
+container runtimes support cgroup v2.
+2. **Testing Workloads**: Verifying that workloads and applications function
+correctly with cgroup v2.
+
+## Further reading
+
+- [Cgroup v2 in
+Kubernetes](https://kubernetes.io/docs/concepts/architecture/cgroups/)
```

Comment:
We use site-relative links where feasible. Additionally, it would be helpful if markdown links were kept on a single line.
```suggestion
- [Cgroup v2 in Kubernetes](/docs/concepts/architecture/cgroups/)
```

Reply:
Can we add the 1.25 cgroup v2 GA blog to further reading as well? 
 - https://kubernetes.io/blog/2022/08/31/cgroupv2-ga-1-25/
```
- [Kubernetes 1.25: cgroup v2 graduates to GA](/blog/2022/08/31/cgroupv2-ga-1-25/)
```